internal/provider/resource_topic.go

```diff
@@ -46,17 +46,27 @@ func (t topicResourceType) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Di
 				Required:            true,
 				Type:                types.StringType,
 				Validators: []tfsdk.AttributeValidator{
-					validation.Compare(validation.OneOf, []string{"JSON", "Binary", "String", "Xml"}),
+					validation.Compare(validation.OneOf, []string{"AVRO", "JSON", "Binary", "String", "Xml"}),
 				},
 			},
+			"key_schema": {
+				MarkdownDescription: "The key type and reference to the schema (if applicable). Read more: https://docs.axual.io/axual/2023.2/self-service/stream-management.html#key-schema",
+				Optional:            true,
+				Type:                types.StringType,
+			},
 			"value_type": {
 				MarkdownDescription: "The value type and reference to the schema (if applicable). Read more: https://docs.axual.io/axual/2023.2/self-service/topic-management.html#value-type",
 				Required:            true,
 				Type:                types.StringType,
 				Validators: []tfsdk.AttributeValidator{
-					validation.Compare(validation.OneOf, []string{"JSON", "Binary", "String", "Xml"}),
+					validation.Compare(validation.OneOf, []string{"AVRO", "JSON", "Binary", "String", "Xml"}),
 				},
 			},
+			"value_schema": {
```

Comment:
The validation here should be: "if there is any value in value_schema, key_type has to be AVRO". Please see my other comment for more details.